store/prefixstore.go

```diff
@@ -60,27 +69,43 @@ func (s prefixStore) Gas(meter GasMeter, config GasConfig) KVStore {
 
 // Implements KVStore
 func (s prefixStore) Iterator(start, end []byte) Iterator {
+	newstart := make([]byte, len(s.prefix))
+	copy(newstart, s.prefix)
```

Comment:
There should be a "cp()" function declared somewhere that makes this easier to read, I suggest we copy that function whereever we need it like this.